packages/components/src/RadioGroup/RadioOption.tsx

```diff
@@ -1,54 +1,65 @@
-import React, { PropsWithChildren } from "react";
+import React, { ReactNode } from "react";
 import { XOR } from "ts-xor";
 import { v1 as uuidv1 } from "uuid";
 import styles from "./RadioGroup.css";
 import { Text } from "../Text";
 
-interface BaseRadioOptions {
+interface BaseRadioOptionProps {
+  /**
+   * The value of the radio button.
+   */
   readonly value: string | number;
+
+  /**
+   * Disables the radio button.
+   */
   readonly disabled?: boolean;
-}
 
-interface RadioOptionsWithDescription extends BaseRadioOptions {
+  /**
+   * Further description of the label.
+   */
+  readonly description?: string;
+
   /**
    * The label to appear beside the radio button.
    */
-  label: string;
+  readonly label?: string;
+
   /**
-   * Further description of the label.
+   * Provide children elements to render a custom label component if `label` is
+   * not provided, or associated additional content with the radio button if
+   * `label` is provided.
+   *
+   * Prefer using `label` and `description` over adding child elements if the
+   * content of either would be text.
```

Comment:
I think we can simplify this and get straight to the point.

```suggestion
   * Render a custom label or additional content below the `label`
   * and `description`.
   *
   * Prefer using `label` and `description` over adding child elements if the
   * content of either would be a string.
```